File: fixtures/scale/python/large/cipherlab/checks/check_decimation.py
from cipherlab.ciphers.decimation import decimation_encrypt, decimation_decrypt
from cipherlab.specs.decimation_spec import decimation_spec
from cipherlab.corpus.corpus_affine import corpus_affine


def check_decimation():
    spec = decimation_spec()
    for text in corpus_affine():
        sealed = decimation_encrypt(text, spec["key"])
        opened = decimation_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "affine"

File: fixtures/scale/ruby/large/lib/check_decimation.rb
require_relative 'decimation_encrypt'
require_relative 'decimation_decrypt'
require_relative 'corpus_affine'

def check_decimation
  corpus_affine.each do |text|
    sealed = decimation_encrypt(text, 12)
    opened = decimation_decrypt(sealed, 12)
    return false if opened != text
  end
  true
end
